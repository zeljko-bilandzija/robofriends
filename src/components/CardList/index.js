import React from "react";

import Card from '../Card';

const cardlist = ({ robots }) => {
    /* if (true) {
        throw new Error('Some error');
    } */
    const cards = robots.map(robot => <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} />);
    
    return (
        <div>
            {cards}
        </div>
    );
};

export default cardlist;
