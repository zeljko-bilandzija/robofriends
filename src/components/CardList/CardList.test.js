import React from 'react';
import { shallow } from 'enzyme';
import CardList from './index';

it('should render CardList Component', () => {
    expect(shallow(<CardList robots={[{id: 1, name: 'Test', email: 'test'}]} />)).toMatchSnapshot();
})