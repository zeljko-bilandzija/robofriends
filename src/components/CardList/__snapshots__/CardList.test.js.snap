// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardList Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <cardlist
    robots={
        Array [
            Object {
              "email": "test",
              "id": 1,
              "name": "Test",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <card
          email="test"
          id={1}
          name="Test"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "test",
          "id": 1,
          "name": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <card
            email="test"
            id={1}
            name="Test"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "test",
            "id": 1,
            "name": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
